lib/linguist/heuristics.yml

```diff
@@ -619,6 +641,11 @@ named_patterns:
   - '^[ \t]*(class|(using[ \t]+)?namespace)\s+\w+'
   - '^[ \t]*(private|public|protected):$'
   - 'std::\w+'
+  euphoria:
+  - '^\s*(namespace)\s+'
```

Comment:
Redundant capturing group, and another redundant quantifier:

~~~suggestion
  - '^\s*namespace\s'